core/src/main/java/org/apache/iceberg/util/Tasks.java

```diff
@@ -226,13 +226,11 @@ private <E extends Exception> boolean runSingleThreaded(
             }
 
             if (stopOnFailure) {
+              threw = true;
```

Comment:
I don't think this the right place to set this? Shouldn't `threw` be set for all cases where we catch an execption, not just the `stopOnFailure` case?

Reply:
Oh, thank you, nice catch. I agree, place `threw` within catch clause is more exact.